osquery/filesystem/windows/fileops.cpp

```diff
@@ -1823,6 +1962,9 @@ Status platformStat(const fs::path& path, WINDOWS_STAT* wfile_stat) {
                         wfile_stat->product_version,
                         wfile_stat->file_version);
 
+  windowsGetOriginalFilename(wstringToString(path.wstring()),
```

Comment:
Could doing this for every call to platformStat impact performance? It may be more performant to have a seperate method to query this information, as even though you might need to open the file handle a second time you could adjust the file table to only call this if the original_filename column is used.

Reply:
It's hard to say. Taking a cursory look at `platformStat` I think it may have some issues. For example, I see it calling `CloseHandle` on an `INVALID_HANDLE_VALUE`.  I suspect the implementation of `platformStat` itself could be optimized.

Reply:
Complexity-wise I think this function is similar to the `windowsGetVersionInfo` function, which is called by `platformStat` just above the call to `windowsGetOriginalFilename` (and unconditionally so). What's more, the sequence of Windows API calls carried out by these functions is similar, too. (Also note that the `for` loops will be trivial in the typical case, with loop iterations <= 1) . So, performance-related issues are not expected. Do you see any specific problems I might have overlooked?

One last thing:
Due to their similar nature, `windowsGetVersionInfo` and `windowsGetOriginalFilename` could be combined into a single function to avoid certain redundancies.

Reply:
Sounds good to me, I don't see any other problems.